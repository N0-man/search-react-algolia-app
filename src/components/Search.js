import React from 'react';

export default class Search extends React.Component {
    constructor(props) {
        super(props)
        this.filterProducts = this.filterProducts.bind(this)
        this.state = {
            products: [
                {name: "Garden Eau de Parfum"},
                {name: "Dionysus small shoulder bag"},
                {name: "Gucci Bloom Nettare Di Fiori"},
                {name: "Growing Pains with Hari Nef"},
                {name: "CRUISE 2019"},
                {name: "The Rajah Bag"},
                {name: "Gucci GG Belt"},
                {name: "Premiere Partie Flacons Vitrifies"}
            ],
            filteredProducts: []
        }
    }

    filterProducts (e) {
        const filteredProducts = this.state.products.filter (product => {
            return product.name.toLowerCase().includes(e.target.value.toLowerCase())
        });
        
        this.setState ( () => ({filteredProducts: filteredProducts}))
    }

    componentDidMount () {
        this.setState ( () => ({
            filteredProducts: this.state.products
        }))
    }

    render () {
        return (
            <div>
                <input type="text" onChange = {this.filterProducts} placeholder="Search Whatever..."/>
                <ul>
                    {this.state.filteredProducts.map ( product => (<li>{product.name}</li>))}
                </ul>
            </div>
        )
    }
}