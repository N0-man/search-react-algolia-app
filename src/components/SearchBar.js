import React from 'react';
import { connect } from 'react-redux';
import { searchProducts } from '../actions/searchProducts'

export const SearchBar = ({searchProducts}) => (
    <div>
        <input className="input-wrapper" type="text" onChange = {(e) => searchProducts(e.target.value)} placeholder="Search Whatever..."/>
    </div>
)

export default connect(undefined, {searchProducts})(SearchBar);