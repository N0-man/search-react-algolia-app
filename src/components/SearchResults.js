import React from "react";
import { connect } from "react-redux";

export const SearchResults = props => {
  if (props.filteredProducts.length === 0)
    return <p className="message">No Results Were Found For '{props.searchTerm}'</p>;

  return (
    <div>
      <p className="message">{props.filteredProducts.length} Results Found For '{props.searchTerm}'</p>;
      <div className="wrapper">
        {props.filteredProducts.map(product => (
          <div className="card">
            <img src={product.img}/>
            <h2>{product.name}</h2>
            <p class="price">$1999.99</p>
            <p>
              <button>Add to Cart</button>
            </p>
          </div>
        ))}
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  filteredProducts: state.filteredProducts,
  searchTerm: state.searchTerm
});

export default connect(mapStateToProps)(SearchResults);
