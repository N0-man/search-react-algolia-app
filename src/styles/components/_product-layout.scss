.wrapper {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
    grid-gap: 20px;
    background-color: #fff;
    color: #444;
    padding: 100px;
    padding-top: 20px;
  }
  
.card {
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    max-width: 300px;
    // margin: auto;
    text-align: center;
    // font-family: arial;
  }

.card img {
    max-width: 295px;
  }

  .price {
    color: grey;
    font-size: 22px;
  }

  .card h2 {
      padding-left: 20px;
      padding-right: 20px;
  }

  .card button {
    border: none;
    outline: 0;
    padding: 12px;
    color: white;
    background-color: #000;
    text-align: center;
    cursor: pointer;
    width: 100%;
    font-size: 18px;
  }

  .card button:hover {
    opacity: 0.7;
  }

  .message {
    margin-left: 100px;
    margin-bottom: 0px;
    margin-top: 30px;
    font-size: 24px;
    font-style: italic;
    color: #676666
  }