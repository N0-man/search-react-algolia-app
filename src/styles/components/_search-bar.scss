
  
  .input-wrapper {
      position: relative;
      margin-left: 100px;
      margin-top: 50px;
  }
  
  input  {
    height: 60px;
    min-width: 50%;
    padding: 0;
    border-radius: 0;
    line-height: 70px;
    background-color: transparent;
    color: #940000;
    font-size: 30px;
    border: none;
    outline: none;
    border-bottom: 3px solid #333333;
  }