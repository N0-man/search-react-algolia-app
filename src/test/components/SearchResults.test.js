import ReactShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import {SearchResults} from '../../components/SearchResults'

test('should render search results correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<SearchResults />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});