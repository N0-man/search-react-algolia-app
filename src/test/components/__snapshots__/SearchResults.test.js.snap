// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render search results correctly 1`] = `
<Context.Consumer>
  [Function]
</Context.Consumer>
`;
